jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpSenderOverHTTP3.java

```diff
@@ -72,12 +72,8 @@ protected void sendHeaders(HttpExchange exchange, ByteBuffer contentBuffer, bool
         else
         {
             String path = relativize(request.getPath());
-            HttpURI uri = HttpURI.build()
-                .scheme(request.getScheme())
-                .host(request.getHost())
-                .port(request.getPort())
-                .path(path)
-                .query(request.getQuery());
+            // URI validations already performed by using the java.net.URI class.
+            HttpURI uri = new HttpURI.Unsafe(request.getScheme(), request.getHost(), request.getPort(), path, request.getQuery(), null);
```

Comment:
A side-effect of this change makes the `HttpURI` immutable.

I wonder if the `MetaData.Request` constructors shouldn't always call `immutable()` on the `HttpURI` parameter they are given?